db/mongodb.go

```diff
@@ -247,3 +257,44 @@ func (m *MongoDB) CreateExtraIndexes(idxs []string) error {
 	slog.Info(fmt.Sprintf("%d index(es) successfully created in the collection %s", len(r), companyTableName))
 	return nil
 }
+
+func (m *MongoDB) Search(q Query) (string, error) {
+	uf := strings.ToUpper(q.Uf)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	coll := m.db.Collection(companyTableName)
+	f := bson.M{"json.uf": uf}
+	if q.Cursor != "" {
+		f["json.cnpj"] = bson.M{"$gt": q.Cursor}
+	}
+	opts := options.Find().
+		SetSort(bson.D{{Key: "json.cnpj", Value: 1}}).
+		SetLimit(1000)
+	c, err := coll.Find(ctx, f, opts)
+	if err != nil {
+		return "", fmt.Errorf("error searching for documents for UF %s: %w", uf, err)
```

Comment:
A gente não está implementando a busca por UF (apenas). A arquitetura é para _busca_. Quando a gente colocar mais campos no `Query` essa mensagem pode ficar obsoleta, então melhor já deixá-la mais alinhada com a arquitetura:

```suggestion
		return "", fmt.Errorf("error running query %#v: %w", q, err)
```